modules.d/90lvm/module-setup.sh

```diff
@@ -90,6 +90,11 @@ install() {
     inst_hook cmdline 30 "$moddir/parse-lvm.sh"
 
     inst_libdir_file "libdevmapper-event-lvm*.so"
+    # libdevmapper-event-lvm2.so.2.xx is not matched above because there is
+    # no .so link, but it is loaded by libdevmapper-event-lvm* above.
+    inst_libdir_file "libdevmapper-event-lvm2.so.*"
```

Comment:
Perhaps mark these optional with "inst_libdir_file -o" ? I understand that dracut should try to install them, but should the installation really fail if these libraries are not available on the host ? Are these hard dependencies ? How come the lvm tests passes without them ?

Reply:
> Perhaps mark these optional with "inst_libdir_file -o" ? 

Without this change, the lvm functionality of dm-eventd is absolutely broken.  Whether or not that functionality is needed in the initramfs I'm not so sure.  but 'dm' is listed in this module as a `depends()`.  I think I would argue that either they should be included, or the 'dm' dependency dropped. Having a working dependency doesn't feel "optional".

> I understand that dracut should try to install them, but should the installation really fail if these libraries are not available on the host 

Is a module-setup able to make a 'dracut' run fail ? My experience is that it is not. I asked about that [here](https://www.spinics.net/lists/linux-initramfs/msg04892.html) .  I probably should have opened an issue, and I can do so.  I wasn't aware that dracut was using github for issues when I posted.

> ? Are these hard dependencies ? How come the lvm tests passes without them ?

I am also not all that perfectly clear on this.  I think the fallout would be that you would not get monitoring from dm-eventd on the initramfs-created devices. The fallout of that would not be recognized in a short-running test.

My experience (from ubuntu and very old centos 7) is that it is worse than that, but maybe this is an ubuntu specific bug.  From experience a dm-eventd gets started in the initramfs that does not have all its dlopene'd dependencies, and it does not get killed before pivot.  The fact that there is a running dm-eventd means the real-root's dm-eventd doesn't get started via systemd and the old one lives on.  My solution locally for that is to just kill dm-eventd before pivot, allowing the real root's systemd unit to start correctly.

Reply:
Thanks for the investigations, this will make review (for others) also earlier.

> [...]  or the 'dm' dependency dropped.

Perhaps we should discuss this possibility based on also on your additional comments..

> My solution locally for that is to just kill dm-eventd before pivot, allowing the real root's systemd unit to start correctly.

> Is a module-setup able to make a 'dracut' run fail ? My experience is that it is not. 

You're right, but what can happen is that lvm dracut module itself might not get included (none of the lvm dracut module parts).

Reply:
> dmeventd is not meant to be run in the initrd. When it's started later, it will monitor LVs activated by the initrd if needed.

So are you saying dmeventd should not be in initramfs at all?  So start with the patch below, and then just remove all other dmeventd things? there is more in modules.d/80lvmthinpool-monitor/start-thinpool-monitor.sh .

```patch
diff --git a/modules.d/90dm/module-setup.sh b/modules.d/90dm/module-setup.sh
index 59467edf..960893cf 100755
--- a/modules.d/90dm/module-setup.sh
+++ b/modules.d/90dm/module-setup.sh
@@ -22,7 +22,6 @@ install() {
         && inst_hook pre-udev 30 "$moddir/dm-pre-udev.sh"
 
     inst_multiple dmsetup
-    inst_multiple -o dmeventd
 
     inst_libdir_file "libdevmapper-event.so*"
 
```

Reply:
Yes, you can remove anything related to dmeventd or monitoring.